chemprop/cli/hpopt.py

```diff
@@ -380,9 +435,18 @@ def tune_model(
 def main(args: Namespace):
     if NO_RAY:
         raise ImportError(
-            "Ray Tune requires ray to be installed. Use 'pip install -U ray[tune]' to install."
+            "Ray Tune requires ray to be installed. Use 'pip install -U ray[tune]' to install ray or use 'pip install -e .[hpopt]' in chemprop folder to install all hpopt relevant packages."
```

Comment:
I made a new conda environment and did `pip install chemprop` and ended up with Python 3.12. I was then unable to `pip install -U ray[tune]` because ray only supports up to 3.11. Maybe we should limit Chemprop to 3.11 or give a note here that ray only works with 3.11.  

The error I got was `ERROR: Could not find a version that satisfies the requirement ray (from versions: none)
ERROR: No matching distribution found for ray`

Reply:
I think if the user follows our documentation at https://chemprop.readthedocs.io/en/main/installation.html#option-1-installing-from-pypi, they wouldn't encounter this problem because we specifically told them to install with python=3.11. But I can update this error message to explain that they should check their environment is currently python=3.11.

```
conda create -n chemprop python=3.11
conda activate chemprop
pip install chemprop
```

Reply:
I have updated the error message to be:

```
    if NO_RAY:
        raise ImportError(
            "Ray Tune requires ray to be installed. If you installed Chemprop from PyPI, make sure that your Python version is 3.11 and use 'pip install -U ray[tune]' to install ray. If you installed from source, use 'pip install -e .[hpopt]' in Chemprop folder to install all hpopt relevant packages."
        )
```